x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/TextSimilarityConfigUpdate.java

```diff
@@ -0,0 +1,216 @@
+            text,
+            textSimilarityConfig.getVocabularyConfig(),
+            tokenizationUpdate == null
+                ? textSimilarityConfig.getTokenization()
+                : tokenizationUpdate.apply(textSimilarityConfig.getTokenization()),
+            Optional.ofNullable(resultsField).orElse(textSimilarityConfig.getResultsField()),
+            Optional.ofNullable(spanScoreFunction).orElse(textSimilarityConfig.getSpanScoreFunction())
+        );
+    }
+
+    @Override
+    public boolean isSupported(InferenceConfig config) {
+        return config instanceof TextSimilarityConfig;
+    }
+
+    @Override
+    public String getResultsField() {
+        return resultsField;
+    }
+
+    @Override
+    public InferenceConfigUpdate.Builder<? extends InferenceConfigUpdate.Builder<?, ?>, ? extends InferenceConfigUpdate> newBuilder() {
+        return new Builder().setText(text).setResultsField(resultsField).setTokenizationUpdate(tokenizationUpdate);
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        TextSimilarityConfigUpdate that = (TextSimilarityConfigUpdate) o;
+        return Objects.equals(text, that.text)
+            && Objects.equals(resultsField, that.resultsField)
+            && Objects.equals(tokenizationUpdate, that.tokenizationUpdate);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(resultsField, tokenizationUpdate, text);
+    }
+
+    public String getText() {
+        return text;
+    }
+
+    public static class Builder implements InferenceConfigUpdate.Builder<TextSimilarityConfigUpdate.Builder, TextSimilarityConfigUpdate> {
+        private String resultsField;
+        private String spanScoreFunction;
+        private TokenizationUpdate tokenizationUpdate;
+        private String text;
+
+        @Override
+        public TextSimilarityConfigUpdate.Builder setResultsField(String resultsField) {
+            this.resultsField = resultsField;
+            return this;
+        }
+
+        public Builder setTokenizationUpdate(TokenizationUpdate tokenizationUpdate) {
+            this.tokenizationUpdate = tokenizationUpdate;
+            return this;
+        }
+
+        public Builder setText(String text) {
+            this.text = text;
+            return this;
+        }
+
+        public Builder setSpanScoreFunction(String spanScoreFunction) {
+            this.spanScoreFunction = spanScoreFunction;
+            return this;
+        }
+
+        @Override
+        public TextSimilarityConfigUpdate build() {
+            return new TextSimilarityConfigUpdate(text, resultsField, tokenizationUpdate, spanScoreFunction);
+        }
+    }
+
+    @Override
+    public Version getMinimalSupportedVersion() {
+        return Version.V_8_4_0;
```

Comment:
Also update to 8.5.0

Reply:
yeah, the PR has been open for a minute :D